### ISSUE TITLE
Password with Special Characters

### ISSUE THREAD

**Issue by DoKraus** [PHP-on-Couch/PHP-on-Couch#95]
Hello there!

I've chosen a password with `!` for my database.
As of your code `urlencode`s it, the given (and accepted) password is not the same as the transmitted one in `CouchClient` (lines 161 / 193).
Is there a specific reason for that?

If I set the credentials for the `CouchClient` by myself it works fine
```php
$this->client = new CouchClient(
    sprintf('https://%s:%s@%s/', $user, $pass, $host), $db_name);
```

Or do I have to set something in `couchdb`'s config file to accept urlencoded credentials?

**Comment by popojargo** [comment]
Hi @DoKraus ,

I might have naively URL encoded the user and password from the URI. 
Instead, I should have used another version (the equivalent of Javascript encoreUriComponent) that doesn't escape the `!` symbol.

I created a pr to fix that #96  which instead of passing the username and password in the URL, passes them through an Authorization header.

**Comment by popojargo** [comment]
Fixed
